config/configstore/model.go

```diff
@@ -319,6 +327,29 @@ func convertAddressesToStringMap(addresses map[config.ContractName]common.Addres
 }
 
 func (nc *NodeConfig) loadFromProtobuf(data *configpb.ConfigData) error {
+	// Nil checks
+	if data == nil {
+		data = new(configpb.ConfigData)
+	}
+	if data.MainIdentity == nil {
```

Comment:
I'm pretty sure this is not the intended behaviour by someone calling the update endpoint.